### ISSUE TITLE
[Feature] Implement a YAML configuration system for experiments

### ISSUE THREAD

**Issue by mythogenesys** [Mohan-CAS-and-hackathons/ecg-denoiser-hackathon#4]
Currently, our main training script (`experiments/run_training.py`) uses command-line arguments to manage hyperparameters like learning rate, loss weights, and model paths. As we add more experiments, this approach becomes cluttered and error-prone.

A much cleaner and more scalable solution is to use configuration files (e.g., in YAML format) to define all the parameters for a specific experiment.

### Your Mission:

Your goal is to refactor the experiment runner scripts to be driven by configuration files. Instead of passing many command-line arguments, a user will simply pass the path to a single config file.

### Core Tasks

1.  **Add a YAML library** to `requirements.txt` (`PyYAML`).
2.  **Create a `configs/` directory** in the project root.
3.  **Create example config files** inside `configs/` for our key experiments.
4.  **Modify `experiments/run_training.py`** to accept a single `--config` argument, load the YAML file, and use its parameters to set up the training run.
5.  **Modify `experiments/run_validation.py`** similarly, so it can load model paths and other settings from a config file.

See the detailed examples below for a clear walkthrough.

### Files to Modify
- `requirements.txt` (to add `pyyaml`)
- `src/experiments/run_training.py`
- `src/experiments/run_validation.py`
- You will create a new `configs/` directory and files within it.

### Before

```python
# This represents the current state of src/experiments/run_training.py
# Notice how many arguments are needed and some values are hardcoded.

import argparse

def train_ecg_model(lr, batch_size, epochs, model_path, use_grad_loss, use_fft_loss):
    print("--- Training ECG Model ---")
    print(f"LR: {lr}, BS: {batch_size}, Epochs: {epochs}")
    print(f"Saving to: {model_path}")
    print(f"Using Gradient Loss: {use_grad_loss}")
    print(f"Using FFT Loss: {use_fft_loss}")
    # ... Actual training logic would go here ...

def main():
    parser = argparse.ArgumentParser(description="Train a model.")
    # PROBLEM: This list of arguments will keep growing!
    parser.add_argument("--lr", type=float, default=1e-4)
    parser.add_argument("--batch_size", type=int, default=32)
    parser.add_argument("--epochs", type=int, default=50)
    parser.add_argument("--model_path", type=str, required=True)
    parser.add_argument("--no-grad-loss", dest='use_grad_loss', action='store_false')
    parser.add_argument("--no-fft-loss", dest='use_fft_loss', action='store_false')
    parser.set_defaults(use_grad_loss=True, use_fft_loss=True)
    
    args = parser.parse_args()

    # The script logic calls the training function with many arguments.
    train_ecg_model(
        lr=args.lr,
        batch_size=args.batch_size,
        epochs=args.epochs,
        model_path=args.model_path,
        use_grad_loss=args.use_grad_loss,
        use_fft_loss=args.use_fft_loss,
    )

if __name__ == "__main__":
    main()

# To run this, you would need a long command:
# python example_before_configs.py --model_path models/my_model.pth --lr 0.005 --no-fft-loss
```

### After

```python
# This represents the NEW version of src/experiments/run_training.py
# It's much cleaner and more scalable.

import argparse
import yaml # Requires PyYAML

# --- First, create a new file: configs/ecg_stpc_full.yaml ---
"""
training_params:
  lr: 0.0001
  batch_size: 32
  epochs: 50

model_params:
  in_channels: 1
  out_channels: 1

loss_weights:
  amplitude: 1.0
  gradient: 0.5
  fft: 0.3

paths:
  data_dir: "data/mit-bih-arrhythmia-database-1.0.0/"
  model_save_path: "models/denoiser_stpc_full_from_config.pth"
"""

def train_ecg_model(config: dict):
    # The training function now takes a single config object.
    print("--- Training ECG Model from Config ---")
    print(f"LR: {config['training_params']['lr']}")
    print(f"Batch Size: {config['training_params']['batch_size']}")
    print(f"Gradient Loss Weight: {config['loss_weights']['gradient']}")
    print(f"Saving to: {config['paths']['model_save_path']}")
    # ... Actual training logic would use values from the config dict ...

def main():
    parser = argparse.ArgumentParser(description="Train a model using a config file.")
    # PROBLEM SOLVED: Only one argument is needed now!
    parser.add_argument(
        "--config", 
        type=str, 
        required=True,
        help="Path to the YAML configuration file for the experiment."
    )
    args = parser.parse_args()

    # Load all parameters from the YAML file.
    with open(args.config, 'r') as f:
        config = yaml.safe_load(f)

    # The script logic is now much cleaner.
    train_ecg_model(config)

if __name__ == "__main__":
    main()

# To run this, the command is simple and powerful:
# python example_after_configs.py --config configs/ecg_stpc_full.yaml
```

### Acceptance Criteria
- [ ] A user can run an entire training experiment with a single command: `python -m experiments.run_training --config configs/eeg_self_supervised.yaml`.
- [ ] Hyperparameters are no longer hardcoded in the training scripts but are loaded from the config file.
- [ ] The new system is documented in the `README.md`'s "Running Experiments" section.
- [ ] All existing experiments can still be run using this new configuration system.

**Comment by Moha4321** [comment]
Hi @mythogenesys this looks like a great feature, i had love to try and implement this. Can you assign it to me?

**Comment by mythogenesys** [comment]
That's great, @Moha4321 i will assign the issue to you
